### PR TITLE
adding donut component <?.x.?>

### PR DESCRIPTION
**Pull request by ryelo**
Donut
screen shot 2018-08-29 at 2 06 10 pm

With tooltip
screen shot 2018-08-29 at 2 06 15 pm

with legend
image

### CODE REVIEW

**Comment by ryelo** [comment]
Added numbers on legend. Still need to give users ability to link sections of the donut.

File: src/Charts/Donut/Donut.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        return (
+            <React.Fragment>
+                <div className='ins-l-donut'>
+                    <div id={this.props.identifier} className={donutClasses}></div>
+                    <div className='ins-c-donut-hole'>
+                        <span className='ins-c-donut-hole--total__number'>{this.props.total}</span>
+                        <span className='ins-c-donut-hole--total__label'>{this.props.totalLabel}</span>
+                    </div>
+                </div>
+                <div className='ins-l-donut__legend'></div>
```
[the remark this answers is not in the document]
Comment (karelhala) [reply src/Charts/Donut/Donut.js]: Given the for loop above you could add div elements and such here and take care of events by d3 I hope I cought all the things in the for loop, if not correct me.
```JSX
<div className='ins-l-donut__legend' ref={ref => {this.legend = ref}}>
  <div class="donut ins-l-donut__legend--item">
  {this.props.values && this.props.values.map(oneItem => (
    <div data-id={oneItem[0]} className="badge-wrapper">
       <span className="badge"></span>
      <span class="badge__label">{oneItem[0]}</span>
      <span class="badge__number">({oneItem[1]})</span>
    </div>
  ))}
  </div>
</div>
```

With this you can just change function `_updateChart` after `if (this.props.withLegend) {` to look like
```JS
if (this.props.withLegend) {
  select(this.legend)
  .on('mouseover', (id) => {
    donut.focus(id);
  })
  .on('mouseout', (id) => {
    donut.revert();
  })
}
```

With this approach you will let react to take care of render (which is much faster than d3's) and d3 to take care of events. You could go even further and add `onMouseOver` and `onMouseOut` to the top element, but you'll have to take care of donut selector and such in these functions, which is somewhat harder.

You'll just have to take care of the background color, but that shouldn't be that hard (just pass color definition to this component and use some default colors).

**Comment by Kinlaw** [comment]
@karelhala++

I need to implement these changes in my Gauge component as well.

**Comment by ryelo** [comment]
@karelhala can you review?

File: src/Charts/Donut/Donut.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            }
+        };
+
+        let donut = c3.generate(donutConfig);
+
+        /* eslint-disable */
+        if (this.props.withLegend) {
+            let id = [];
+            let num = [];
+            for (let i = 0; i < this.props.values.length; i++) {
```
Comment (karelhala) [inline src/Charts/Donut/Donut.js]: This for loop is not needed anymore

**Comment by ryelo** [comment]
@karelhala one last time (hopefully)!

**Pull request merged**